### ISSUE TITLE
add to the wounded the display of the time of blood loss (not the skull, but the timer)

### ISSUE THREAD

**Issue by topden** [diwako/armor_plates_system#100]
add to the wounded the display of the time of blood loss (not the skull, but the timer)
pictures are cool, but a simple timer to understand would be better. Here's what it looks like for me
http://joxi.ru/DrlBgy0CKwgdz2
http://joxi.ru/VrwzEjdcoPegMA
http://joxi.ru/n2YXgExCkvz0vm
http://joxi.ru/82QLB8Kc85x4Om
increasing the icon of the wounded and the timer in the sight very much blocks the view

I also want to note that disabling your system in the settings does not allow you to play with vanilla revive. The game always works in respawn mode. Is it possible to change this without deleting the mod?

**Comment by diwako** [comment]
I am sorry, I do not understand? These icons in those screenshots are not from this mod?

**Comment by topden** [comment]
no. this is just my example. these icons and the timer look the same at different distances and in scopes. do not increase. it does not hurt the eyes and does not block the view. I did this for the vanilla revive with a simple event handler. Maybe you want to change your choice. it's just not very convenient when the icon and the timer constantly decrease or increase in different situations.
`addMissionEventHandler ["Draw3D",{
		private _distance = [0, 500] select (player getUnitTrait "medic");
		private _3d_icon_size   = 1;
		private _3d_icon_color  = [1,0,0,1];
		private _3d_icon_death_color = [7, 35, 219, 1];
		private _text  = "";
		private _allPlayers = [];
		private ["_timer"];

		if (lifeState player == "INCAPACITATED") then {
			private _bleed = player getVariable "#rev_bleed";
			private _damage = player getVariable "#rev_damage";
			private _timer = floor (bis_revive_bleedOutDuration * (1 - _bleed - _damage));
			player setVariable ["tr_timer", _timer, true];
		};
		{
			_pos = ASLToAGL getPosASL _x;
			if (_x distance player < _distance && {lifeState _x == "INCAPACITATED"}) then {
				_timer = _x getVariable ["tr_timer", 300];
				drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Revive\overlayIcons\u100_ca.paa", _3d_icon_color, [_pos # 0, _pos # 1, (_pos # 2) + 0.2],_3d_icon_size,_3d_icon_size,0,format ["%1",_timer],1,0.05,"RobotoCondensedBold"];
			};
			if (_x distance player < _distance && {lifeState _x == "DEAD-RESPAWN"}) then {
				drawIcon3D["\a3\ui_f\data\IGUI\Cfg\Revive\overlayIconsGroup\f100_ca.paa",_3d_icon_death_color,[_pos # 0,_pos # 1,(_pos # 2) + 0.2],_3d_icon_size,_3d_icon_size,0,format[_text],1,0.05];
			};
		} forEach allPlayers;
	}
];`

**Comment by topden** [comment]
http://joxi.ru/krDRzKMHGbjj52
http://joxi.ru/EA44LwoiXQEEBA
here is an example of **your icon size** adjustment and **MY timer** in your mod. you see, it does not block the view, while the icons increase greatly when zoomed in

**Comment by topden** [comment]
this is how i made icons and timer for myself in your mod
https://youtu.be/_cdLAHb_kqg

**Comment by diwako** [comment]
I ma all up for the change if it can be set inside the Addon Options menu.

**Comment by topden** [comment]
fnc_drawDownedUnitIndicator.sqf
------------
#include "script_component.hpp"

private _size = GVAR(showDownedUnitIndicatorSize) * ((call CBA_fnc_getFov) select 1);
private _camPos = positionCameraToWorld [0, 0, 0];
private _color = [
    profileNamespace getvariable ['igui_error_RGB_R', 1],
    profileNamespace getvariable ['igui_error_RGB_G', 0.1],
    profileNamespace getvariable ['igui_error_RGB_B', 0.1],
    profileNamespace getvariable ['igui_error_RGB_A', 1]
];

private _drawParams = [
    [GVAR(showDownedUnitIndicatorRange), GVAR(bleedoutTimerRange), 2], // rifleman values
    [GVAR(showDownedUnitIndicatorRangeMedic), GVAR(bleedoutTimerRangeMedic), 1] // medic values
] select ((call CBA_fnc_currentUnit) getUnitTrait "Medic");
_drawParams params ["_downedMaxRange", "_bleedoutMaxRange", "_visibleSetting"];

{
    private _distance = _x distance _camPos;
	
	private _pos = ASLToAGL getPosASL _x;
	private _3d_icon_death_color = [7, 35, 219, 1];
	
    if ( _distance > _downedMaxRange ) then {continue};

    // private _sizeFinal = linearConversion [0, _downedMaxRange, _distance, _size, _size/10, false];
	
    drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Revive\overlayIcons\u100_ca.paa", _color, [_pos # 0,_pos # 1,(_pos # 2) + 0.2],1,1,0,format[_text],1,0.05];
	
    if (_distance > _bleedoutMaxRange || {GVAR(visibleBleedoutTimer) < _visibleSetting}) then {continue};
	
    private _timeRemaining = round ((_x getVariable [QGVAR(bleedoutKillTime), -1]) - cba_missionTime);
	
    if (_timeRemaining >= 0) then {
        drawIcon3D ["", GVAR(bleedoutTimerColor), [_pos # 0, _pos # 1, (_pos # 2) + 0.1], 1, 1, 0,format ["%1",_timeRemaining],1,0.05,"RobotoCondensedBold"];
    };
				
} forEach (GVAR(downedUnitIndicatorDrawCache) select {_x getVariable [QGVAR(unconscious), false]});

------------
I did it myself. Highlight what you changed in bold. I removed the ability to adjust the size of the icon. left only the timer color adjustment. (I also removed the size adjustment from the mod settings). I'm sure you can do it better or more correctly, since I'm not a scripter, I'm just learning) Maybe you will like this idea, or maybe you will do it better)
